### PR TITLE
Refactor addon-jest to use a parameter-based pattern

### PR DESCRIPTION
**Pull request by tmeasday**
For https://github.com/storybooks/storybook/issues/3625 -- this one was a little different as the current API doesn't quite fit the old pattern. But I think I've brought it as close as possible to fitting the parameters philosophy.

### CODE REVIEW

File: addons/jest/README.md [hunk 1]
```diff
@@ -83,48 +86,55 @@ import results from '../.jest-test-results.json';
...
+    () => <div>Jest results in storybook</div>
+  );
 ```
 
 ### withTests(options)
 
-- **options.results**: OBJECT jest output results. *mandatory*
-- **filesExt**: STRING test file extention. *optionnal*. This allow you to write "MyComponent" and not "MyComponent.test.js". It will be used as regex to find your file results. Default value is `((\\.specs?)|(\\.tests?))?(\\.js)?$`. That mean it will match: MyComponent.js, MyComponent.test.js, MyComponent.tests.js, MyComponent.spec.js, MyComponent.specs.js...
+* **options.results**: OBJECT jest output results. _mandatory_
+* **filesExt**: STRING test file extention. _optionnal_. This allow you to write "MyComponent" and not "MyComponent.test.js". It will be used as regex to find your file results. Default value is `((\\.specs?)|(\\.tests?))?(\\.js)?$`. That mean it will match: MyComponent.js, MyComponent.test.js, MyComponent.tests.js, MyComponent.spec.js, MyComponent.specs.js...
```
Comment (shilman) [inline addons/jest/README.md]: optionnal => optional 😄

**Review by shilman — APPROVED** [review]
LGTM w/ small typo

File: addons/jest/README.md (hunk 1 above)
Comment (tmeasday) [reply addons/jest/README.md]: That was an existing line ;)

File: addons/jest/src/index.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import addons from '@storybook/addons';
+import deprecate from 'util-deprecate';
 
 const findTestResults = (testFiles, jestTestResults, jestTestFilesExt) =>
-  testFiles.map(name => {
+  [].concat(testFiles).map(name => {
```
Comment (renaudtertrais) [inline addons/jest/src/index.js]: why ? `map()` is immutable no ?

**Comment by shilman** [comment]
@tmeasday looks like the chromatic snapshot is failing?

**Review by shilman — CHANGES_REQUESTED** [review]
chromatic snapshot failing?

File: addons/jest/src/index.js (hunk 2 above)
Comment (tmeasday) [reply addons/jest/src/index.js]: `[].concat()` is a way to "ensure array" -- `[].concat('a') == [].concat(['a'])`

Comment (tmeasday) [reply addons/jest/src/index.js]: So it lets the user set `{ jest: 'filename' }` or `{ jest: ['filename', 'other-filename'] }` which is natural.

**Comment by tmeasday** [comment]
> chromatic snapshot failing?

Arggh, looks like this bug has reared its head again.

**Comment by tmeasday** [comment]
@renaudtertrais are you OK with the changed API/readme here?

File: addons/jest/src/index.js (hunk 2 above)
Comment (Hypnosphi) [reply addons/jest/src/index.js]: `Array.from(testFiles)` is a more explicit way to do the same

**Review by renaudtertrais — APPROVED** [review]

**Review by shilman — APPROVED** [review]

File: addons/jest/src/index.js [hunk 3]
```diff
@@ -27,9 +28,27 @@ export const withTests = userOptions => {
...
+
+    const [
+      story,
+      {
+        kind,
+        parameters: { jest: testFiles },
+      },
+    ] = args;
+
+    if (testFiles && !testFiles.skip) {
```
Comment (Keraito) [inline addons/jest/src/index.js]: Where does this `testFiles.skip` come from? Is this the same as the `disable` flag that we have in the other parameterized addons?

**Review by Keraito — APPROVED** [review]

File: addons/jest/src/index.js (hunk 3 above)
Comment (Keraito) [reply addons/jest/src/index.js]: Poking @tmeasday just in case you missed this comment, since this is part of the 4.0 release.

Comment (tmeasday) [reply addons/jest/src/index.js]: Thanks @Keraito -- I have not forgotten, just getting around to this, sorry!

**Pull request merged**